### ISSUE TITLE
UI change to require extra clicks?

### ISSUE THREAD

**Issue by ckotwn** [gbif/ipt#1946]
I've tried to get used to, but not sure if this is only me. The current dataset overview page has folded many functional buttons under the ⠇menu. This not only make daily activities more cumbersome, but also add difficulties in training scenarios.

I do appreciate that IPT UI has become more modern and sleek. Though please do keep function buttons as prominent as possible, such as edit, delete, preview, public/private, publish and register. There are not that many functions that need folding after all on the page IMO. The border line already adds much clarity.

In short, I am suggesting those folded functions under ⠇menu would be released as visible buttons/links upon page loading. Thanks for considering and saving trainees some trouble navigating features.

**Comment by mike-podolskiy90** [comment]
@ckotwn Thank you very much for your feedback.
Intention was to simplify the interface and improve the mobile version. If the current view is causing problems we can of course change that by replacing the three-dot menus for the desktop version with real buttons. But I would also like to hear more opinions on that.
I'll post screenshots of possible solutions here.

**Comment by ckotwn** [comment]
Thanks. I am not sure whether IPT should be mobile-first. And I hate macOS system prefs forcing me to do more clicks and scrolls. Oh, and in terms of procedure, if there were no hearing-opinions before implementing the 3-dot menu, I think it may be a bit unfair for the reverse? ;) My general concern of IPT UI changes is the associated efforts needed to update the materials (still lagged and being voiced by trainees) and costs when we need to help users navigate the changes. I know, I know, they are not the end of the world, they are just annoying at worst.

Anyway please post possible solutions here and I am keen to work with you to the most sensible one. Others please chime in!

Great job on the sleekness again, @mike-podolskiy90 !

**Comment by SophiePamerlon** [comment]
Hi, I just had a look at the latest version of the IPT and here are my thoughts on it:
- I like the new UI & colour scheme, I think it makes it visually clearer for the user(s)
- I'm still getting used to the  ⠇menu but I don't think it will be 'worse' than previous upgrades for training materials, we regularly have to update our screencaps anyway :)
- The extra-clicks might indeed be a bother for some users though? But nothing that can't be explained/demo-ed, especially as it's still possible to take screenshots with the hidden option appearing on screen (just tested it with the Firefox screenshot tool)
- One detail (bug?) I noticed is that we cannot access the source file page (with technical details such as the encoding format or separator option) anymore? I can only see the "+ Add" option in the  ⠇menu for the source file section
- I agree with @ckotwn that the IPT is mainly used on desktop, so adjustments for the mobile version should not privileged in my opinion
- thanks for your great work @mike-podolskiy90 ! :)

**Comment by mike-podolskiy90** [comment]
@SophiePamerlon Thank you very much for your feedback, I appreciate it.

Editing sources/mappings should be accesible by just clicking on the element. Should we also add an explicit button for that?

Registration is not accesible anymore indeed (it's still accesible on the main resource page though at the top of the page)

**Comment by dbloom** [comment]
I also have just passed through the newest release of the IPT. Some additional thoughts:

1. I agree with @ckotwn and @SophiePamerlon that the extra clicks will be annoying. As a user who once or twice a year has to update 100+ resources manually on an IPT, these extra clicks will grow tiresome. I know that my case may not be the norm, but even when setting up multiple resources in the same session those extra clicks make a difference. As a user of both Mac and PC platforms I have a sense that this is the sort of thing that will pass once people use the U/I for a couple of weeks. The really annoying stuff, if anything, will rise to the top and find its way to GitHub.

2. The color scheme is a nice idea, especially in the tables one the home and manage resource tabs, BUT the execution makes the interface look schizophrenic and messy when there are multiple colors in multiple columns. I might suggest picking a color palette for each columns (e.g, various shades of blue/purple in the Type column, shades of red/orange in Subtype and, shades of green/yellow in Visibility). As it is now the reds and oranges in different columns pull the eye in 10 different directions and it is distracting when attempting to read content and navigate the lists.  E.g., this screen view has my eyes shooting all over the place between red alert text, Subtype status and Visibility. 
![Screen Shot 2023-02-09 at 8 55 49 AM](https://user-images.githubusercontent.com/942447/217884115-dd7fc6d0-231e-4a00-a67f-e71036c9a087.png)

Giving those colors some structure and organization, and possibly even using slightly muted tones would help significantly.

3. When working with a resource via the overview page those "three dots" over on the right, are difficult to see - once you know they are there, you see them, but even I was wandering around trying to find an "edit" option. Like the extra click issue, the regular need to scroll to the right side, when everything else is over on the left is annoying. Could the dot menu be located in proximity to the rest of the content - perhaps just to the right of the section headers (e.g., Source Data, Darwin Core Mappings)? It just feels like the dot menu offended someone at GBIF has has been banished to the right-hand margin until it agrees to apologize.

Re: the extra clicks and missing buttons - The first place I looked to find the option Add a new source file or to Edit the mappings was not the menu on the far right, it was the dot menu in the existing source and mapping files/links. It was a bit surprising not to find either Add or Edit right there. That will freak some people out. Relocating the primary dot menu over might help.

4. Similarly, the boxes that contain the files or references to source files, mappings and other info are a bit confusing. Some of those boxes are clickable and take the user to a new page or view. Others don't do anything and leave the user wondering if something is supposed to happen. For example, clicking on a source file, takes you to the source overview, but clicking on the name of a resource manager or on the Registration does nothing even though the interaction with all of the boxes is the same (Note the width of the registration box is double that of every other box on the page). Perhaps those boxes that don't result in a change can be viewed more like the content in the Visibility section to eliminate any expectations that clicking will bring users to the Promised Land?

5. Finally, I already miss the "View" button in the current version. I know it's in the menu, but, but, but, it was so easy to click.... right there.... waving at me.... Saying "click me!"    I'll survive it. Just sayin'.

@mike-podolskiy90 This still looks amazing and I have enjoyed watching the evolution of the tool over the years. The look overall is much cleaning and more modern, which I think reflects well on your work and GBIF's image. Nothing listed above is a deal-breaker for me, although I think there can be some improvements. Mind you, I'll never be a @SophiePamerlon, but I can try.

**Comment by mike-podolskiy90** [comment]
@dbloom Thank you for your comments

**Comment by rondlg** [comment]
Hi,

the updated UI fine. I'm not sure how often folks do any real work on a mobile device but that might just be me. I certainly can't foresee training anyone to do that, but it's fun to have. 

I am sure all the accessibility checking has been done, just a couple of possible niggles/pet peeves: 

1) I'm a little concerned about the contrast of text and background for readability. Especially on the pastel tags.
2) The 3 dot menus don't include all the options. Most obviously for me the "edit" option, leaving you to click around to eventually realise that the box around the text shades thereby making it a button!

Otherwise, great job.

**Comment by mike-podolskiy90** [comment]
@rondlg Thank you for your feedback

**Comment by mike-podolskiy90** [comment]
Thanks for the great feedback on this.
I think I've managed to apply most of the desired changes to https://ipt.gbif-uat.org.
We're still working on the color scheme and look of the pills, but here are some ideas:

![Screenshot 2023-02-13 at 15 41 02](https://user-images.githubusercontent.com/24795387/218490795-3df9a805-68a4-4e59-96fb-1b62df463cf8.png)

![Screenshot 2023-02-13 at 15 43 34](https://user-images.githubusercontent.com/24795387/218490918-743924c1-e651-4b3a-a953-28456ec37581.png)

![Screenshot 2023-02-13 at 15 47 31](https://user-images.githubusercontent.com/24795387/218490968-956fe0bd-1767-4247-a69d-56e8fb62e190.png)

**Comment by peterdesmet** [comment]
Can I suggest the following colours for the badges:

<img width="362" alt="Screenshot 2023-02-15 at 10 26 32" src="https://user-images.githubusercontent.com/600993/218987536-46b66963-1959-4ad4-836f-99bd6f27f570.png">

**Dataset type and subtype**: I would treat these as tags of the same colour, so they are given the same visual weight. I opted for regular text color (`--color-gbif-secondary`) for text and border + a white background.

**Visibility**: I would treat these as ordered categories. I opted for the same design as (hoovered) buttons in the IPT, using the color for text and border + a background with the color at opacity `0.04`:
- **private**: either the red `--color-gbif-danger` or the milder yellow `--color-gbif-warning`. I think I prefer the `warning` one.
- **public**: `--color-gbif-primary`
- **registered**: `--color-gbif-primary` with full opacity background and white text

<details>
<summary>CSS</summary>
<pre>
   .type {
      /* --color-gbif-secondary */
      color: rgb(78,86,95);
      border: 1px solid rgb(78,86,95);
      background: white;
   }   
   .private-danger {
      /* --color-gbif-danger */
      color: rgb(227,99,112);
      border: 1px solid rgb(227,99,112);
      background: rgba(227,99,112, 0.04);
   }
   .private-warning {
      /* --color-gbif-warning */
      color: rgb(255,193,8);
      border: 1px solid rgb(255,193,8);
      background: rgb(255,193,8, 0.04);
   }
   .public {
      /* --color-gbif-primary */
      color: rgb(97,168,97);
      border: 1px solid rgb(97,168,97);
      background: rgba(97,168,97, 0.04);
   }
   .registered {
      /* --color-gbif-primary */
      color: white;
      border: 1px solid rgb(97,168,97);
      background: rgb(97,168,97);
   }
</pre>
</details>

**Comment by mike-podolskiy90** [comment]
@peterdesmet Thank you for your suggestions!

We're currently finalising home pages, they'll be looking something like this
![Screenshot 2023-02-15 at 14 14 10](https://user-images.githubusercontent.com/24795387/219037345-9f91be70-fae5-43ba-884d-dea915a9d15e.png)

https://ipt.gbif-uat.org/

**Comment by ckotwn** [comment]
@mike-podolskiy90 thanks for the update. I've had a look at the resource overview. It looks great that the functions are not folded. To make it visually more relevant, would you consider enlarge the button and place it at the top right of `<div class="mt-4"></>` hence below the heading and after the first line of the paragraph? (It looks too far out to me, not sure others though).
 
<img width="1006" alt="Screenshot 2023-02-16 at 13 22 35" src="https://user-images.githubusercontent.com/637631/219277563-02a2a335-fef5-4e21-9f57-d2c34700ac89.png">

**Comment by dbloom** [comment]
@mike-podolskiy90 That last screen cap looks much better. Thank you for taking the time to smooth the colors. It is much easier to view and navigate now.

I'd like to look through the UAT, but my login doesn't appear to be functional. I will check with you separately about that.

**Comment by mike-podolskiy90** [comment]
Thank you all for the suggestions. We've applied most of them.
IPT is almost ready and available to reviewing.

**Comment by mike-podolskiy90** [comment]
@ckotwn @dbloom @SophiePamerlon @rondlg @peterdesmet 
If you happen to have time please give it a try at [ipt.gbif-uat.org](https://ipt.gbif-uat.org/)

**Comment by ckotwn** [comment]
Thanks, Mike. I agree with dbloom that the colors on the resource list are much reasonable now. For what I suggest about making function buttons more prominent with bigger text and color buttons, will you consider to implement? I will certainly love IPT more if that happens :)

**Comment by peterdesmet** [comment]
Looks good to me! Labels on homepage are great. Showing the buttons rather than the `...` is an improvement. Agree with @ckotwn that the buttons are not very prominent, but it doesn't bother me.

A very minor nitpick: I would increase the spacing between icon and text for better readability.

<img width="93" alt="current" src="https://user-images.githubusercontent.com/600993/220915171-66eea916-8885-4542-951b-c7efae0957a2.png">
to:
<img width="119" alt="new" src="https://user-images.githubusercontent.com/600993/220915196-b45159cf-bfad-4528-955c-df26a64bac88.png">

I did this by adding `margin-right: 0.2rem` to `.overview-action-button-icon`

**Comment by mike-podolskiy90** [comment]
Thank you @peterdesmet , I'll apply that.

@ckotwn Thank you for the suggestion, but looks like it just doesn't look nice. I'd prefer to keep as it is

![Screenshot 2023-02-23 at 16 48 58](https://user-images.githubusercontent.com/24795387/220959324-d4af9408-ec1e-4c96-90de-a1cf3c80283e.png)

**Comment by albenson-usgs** [comment]
I don't think I have a login to the UAT to see what the buttons look like right now but maybe a compromise is to make the buttons bigger but keep them transparent (i.e. not green)?

**Comment by mike-podolskiy90** [comment]
@albenson-usgs  How it looks right now:

![Screenshot 2023-02-23 at 17 00 06](https://user-images.githubusercontent.com/24795387/220961903-b4bc173a-070f-41ca-b461-0059f2409403.png)

I can create an account for you there if you wish

**Comment by albenson-usgs** [comment]
Yes, please create me an account if that's not too much trouble. Thank you!

**Comment by albenson-usgs** [comment]
After logging in and working a bit, I do agree the buttons are a little small. I still think a good compromise would be to keep them the current color combination but make them a bit bigger.

**Comment by mike-podolskiy90** [comment]
![Screenshot 2023-02-23 at 17 49 05](https://user-images.githubusercontent.com/24795387/220974477-fa6b6f05-2d03-411e-957e-7b3995df73f6.png)

**Comment by ckotwn** [comment]
> @ckotwn Thank you for the suggestion, but looks like it just doesn't look nice. I'd prefer to keep as it is

Thanks for trying out. I can see that the bold button could steal attention from the overall light-coloured visuals.

**Comment by mike-podolskiy90** [comment]
Thank you all for your input. I'm closing the issue.
